src/components/common/OfficeAddresses.vue

```diff
@@ -221,242 +228,169 @@ import { Actions } from '@/enums'
     'mailing-address': BaseAddress
   },
   computed: {
+    ...mapState(['entityIncNo']),
     ...mapGetters(['isBComp'])
   }
 })
 export default class OfficeAddresses extends Mixins(CommonMixin) {
-  // Local definition of computed property for static type checking.
+  // Local definitions of computed properties for static type checking.
+  // NB: use non-null assertion operator to allow use before assignment
   readonly isBComp!: boolean
+  readonly entityIncNo!: string
 
-  /**
-   * Indicates whether this component should be enabled or not.
-   */
+  /** Indicates whether this component should be enabled or not. */
   @Prop({ default: true })
   readonly componentEnabled: boolean
 
   /**
-   * Addresses object from the parent page.
-   * If this is null then this is a new filing; otherwise these are the addresses from a draft filing.
-   * This will be emitted back to the parent page when the addresses are updated.
-   */
-  @Prop({ default: null })
-  readonly addresses: BcorpAddressIf | null
-
-  /**
-   * Registered Office address object passed in from the parent which is pulled from store.
-   * This address is used as a baseline address in the event the user wants to reset the changes before submitting.
-   */
-  @Prop({ default: null })
-  private registeredAddress: BaseAddressObjIF
-
-  /**
-   * Records Office address object passed in from the parent which is pulled from store.
-   * This address is used as a baseline address in the event the user wants to reset the changes before submitting.
+   * Addresses object from parent page, used to set our working addresses (4 properties below).
+   * As needed, we emit this back to parent (ie, update/sync).
    */
-  @Prop({ default: null })
-  private recordsAddress: BaseAddressObjIF
+  @Prop({ default: () => {} })
+  readonly addresses: RegRecAddressesIF
```

Comment:
This prop object is now correctly initialized as an object (ie, so we can test "if addresses.something", which you can't do if it's null).
This needed a change to check "ifEmpty" in a few places.